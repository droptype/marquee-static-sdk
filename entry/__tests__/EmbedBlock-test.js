const React = require('react')
const EmbedBlock = require('../EmbedBlock')
const renderer = require('react-test-renderer')

const SAMPLE_BLOCK = require('./SAMPLE_ENTRY.json').content.find( block => 'embed' === block.type)

describe('EmbedBlock', () => {
    describe('parseVideoURL', () => {
        it('handles YouTube links', () => {
            const { parseVideoURL } = require('../EmbedBlock')
            
            expect(
                parseVideoURL('https://www.youtube.com/watch?v=dQw4w9WgXcQ')
            ).toEqual('//www.youtube.com/embed/dQw4w9WgXcQ?modestbranding=1')

            expect(
                parseVideoURL('https://youtu.be/dQw4w9WgXcQ')
            ).toEqual('//www.youtube.com/embed/dQw4w9WgXcQ?modestbranding=1')
        })

        it('handles Vimeo links', () => {
            const { parseVideoURL } = require('../EmbedBlock')
            
            expect(
                parseVideoURL('https://vimeo.com/168590856')
            ).toEqual('//player.vimeo.com/video/168590856')

            expect(
                parseVideoURL('https://player.vimeo.com/video/168590856')
            ).toEqual('https://player.vimeo.com/video/168590856')
        })
    })

    it('renders', () => {
        const component = renderer.create(
            React.createElement(EmbedBlock, { block: SAMPLE_BLOCK })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })

    it('renders plain', () => {
        const component = renderer.create(
            React.createElement(EmbedBlock, { block: SAMPLE_BLOCK, plain: true })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })

})
