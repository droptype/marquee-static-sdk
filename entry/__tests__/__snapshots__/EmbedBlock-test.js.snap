// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedBlock renders 1`] = `
<figure
  className={
    Classes {
      "_groups": Object {
        "position": "center",
        "size": "medium",
        "source": "embedly_html",
      },
      "_initial": Array [
        "Block",
        "EmbedBlock",
      ],
      "_switches": Object {},
    }
  }
  id="embed_47506d9d71e94c1b864d0f8b3b332b0b"
>
  <div
    className="_Content"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<blockquote class=\\"twitter-tweet\\"><a href=\\"https://twitter.com/marquee/status/832694046069960704\\"></a></blockquote><script async src=\\"//platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script>",
        }
      }
    />
    <figcaption
      className="_Caption"
    >
      <p
        className="_CaptionText"
      />
      <p
        className="_Credit"
      />
    </figcaption>
  </div>
</figure>
`;

exports[`EmbedBlock renders plain 1`] = `
<figure>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<blockquote class=\\"twitter-tweet\\"><a href=\\"https://twitter.com/marquee/status/832694046069960704\\"></a></blockquote><script async src=\\"//platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script>",
      }
    }
  />
  <figcaption
    className={null}
  >
    <p
      className={null}
    />
    <p
      className={null}
    />
  </figcaption>
</figure>
`;
